pallets/public-credentials/src/lib.rs

```diff
@@ -585,6 +594,35 @@ pub mod pallet {
 				}
 			})
 		}
+
+		#[cfg(any(feature = "try-runtime", test))]
+		pub fn do_try_state() -> Result<(), &'static str> {
+			Credentials::<T>::iter().try_for_each(
+				|(subject_id, credential_id, entry)| -> Result<(), &'static str> {
+					ensure!(
+						CredentialSubjects::<T>::contains_key(&credential_id),
+						"Unknown credential subject"
+					);
+
+					ensure!(
+						CredentialSubjects::<T>::get(&credential_id) == Some(subject_id),
```

Comment:
This check also is sufficient, and you can remove the previous one since they check for the same thing.